### ISSUE TITLE
Shouldn't a warning result in a return code ($?) higher then zero?

### ISSUE THREAD

**Issue by gerardjp** [wp-cli/wp-cli#1430]
Hi All,

I'm working on some wrapper software and noticed a wrong return code on the commandline:

``` bash
root@srvyy:~# wp --allow-root --path=/var/www/www.test.nl plugin deactivate p3-profile
Warning: The 'p3-profile' plugin could not be found.
(nb: the plugin name is actually p3-profiler)
root@srvyy:~# echo $?
0
root@srvyy:~# ls /not-here
ls: cannot access /not-here: No such file or directory
root@srvyy:~# echo $?
2
root@srvyy:~#
```

Shouldn't it produce an errorlevel higher then zero or am I missing something?

Thanx a lot.

Kind regards,

Gerard.

**Comment by gerardjp** [comment]
Working on a Wordpress plugin that enables some new wp-cli commands  .. I just noticed in the API docs:

```WP_CLI::error( $msg ) - prints an error on STDERR and exits with code 1```

(ref: https://github.com/wp-cli/wp-cli/wiki/API)

This suggests an actual return code ... Is the documentation erroneous or did I mis somehting :smiley: 

Ta!

**Comment by fabiokr** [comment]
How about a `--fail` option that would force failure?

**Comment by ypid** [comment]
```Shell
wp plugin uninstall $active_plugin
```

should return something else than 0 …

**Comment by danielbachhuber** [comment]
For those subscribed to this thread, I've posted a RFC on changing the behavior #3577

**Comment by danielbachhuber** [comment]
This is being addressed by #3577
